### PR TITLE
adding debounce to update component tree after all updates are done

### PR DESCRIPTION
**Pull request by sumitarora**

### CODE REVIEW

File: frontend/batarangle.ts [hunk 1]
```diff
@@ -48,8 +49,10 @@ class App {
 
     this.componentDataStore.dataStream
       .map(({ componentData }: any) => componentData)
+      .debounce((x) => {
```
Comment (igor-ka) [inline frontend/batarangle.ts]: @sumitarora can't we just use ```.debounce(500)```?

Comment (sumitarora) [inline frontend/batarangle.ts]: `.debounce(500)` just work with event's not data stream with updated version

Comment (igor-ka) [inline frontend/batarangle.ts]: :+1:

**Comment by igor-ka** [comment]
@sumitarora LGTM :shipit:

**Pull request merged**